File: programmers/java/lev0/빈 배열에 추가, 삭제하기.java
import java.util.Stack;
class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
         Stack<Integer> stack = new Stack<>();
       
       for(int i=0;i<flag.length;i++){
            if(flag[i]){
                for(int j=0;j<arr[i]*2;j++)
                stack.push(arr[i]);
            }else{
                for(int j=0;j<arr[i];j++)
                stack.pop();
            }
       }

       int[] answer = new int[stack.size()];

       for (int i = 0; i<stack.size(); i++) {
           answer[i] = stack.get(i);
       }
        return answer;
    }
}

File: programmers/js/lev0/빈 배열에 추가, 삭제하기.js
function solution(arr, flag) {
  let answer = [];

  flag.forEach((el, i) => {
    if (el) {
      for (let j = 0; j < arr[i] * 2; j++) answer.push(arr[i]);
    } else {
      for (let j = 0; j < arr[i]; j++) answer.pop();
    }
  });
  return answer;

  //   return arr.reduce(
  //     (prev, num, i) => (flag[i] ? [...prev, ...new Array(num * 2).fill(num)] : prev.slice(0, -num)),
  //     [],
  //   );
}
